build/__tools__/run-tests.sh

```diff
@@ -284,5 +299,67 @@ if [ $? == 0 ]; then
     die mock contract test succeeded though it should have failed
 fi
 
+#-----------------------------------------
+yell run tests with the eservice database 
+#------------------------------------------
+
+say run various pdo scripts - test-request, test-contract, create, update, shell - using database
+try pdo-test-request --eservice-name e1 --logfile $PDO_HOME/logs/client.log --loglevel info  
+try pdo-test-contract --contract integer-key --eservice-name e2 --logfile $PDO_HOME/logs/client.log --loglevel info 
+
+# make sure we have the necessary files in place
+CONFIG_FILE=${PDO_HOME}/etc/pcontract.toml
+if [ ! -f ${CONFIG_FILE} ]; then
+    die missing client configuration file, ${CONFIG_FILE}
+fi
+
+CONTRACT_FILE=${PDO_HOME}/contracts/_mock-contract.scm
+if [ ! -f ${CONTRACT_FILE} ]; then
+    die missing contract source file, ${CONTRACT_FILE}
+fi
+
+try pdo-create --config ${CONFIG_FILE} --ledger ${PDO_LEDGER_URL} \
+     --logfile $PDO_HOME/logs/client.log --loglevel info \
+    --identity user1 --save-file ${SAVE_FILE} \
+    --contract mock-contract --source _mock-contract.scm --eservice-name e1 e2 e3 
+
+# this will invoke the increment operation 5 times on each enclave round robin
+# fashion; the objective of this test is to ensure that the client touches
+# multiple, independent enclave services and pushes missing state correctly
+declare -i pcontract_es=3 #.see ../opt/pdo/etc/template/pcontract.toml
+declare -i n=$((NUM_SERVICES*pcontract_es)) e v value
+for v in $(seq 1 ${n}) ; do
+    e=$((v % pcontract_es + 1))
+    value=$(pdo-update --config ${CONFIG_FILE} --ledger ${PDO_LEDGER_URL} \
+                       --eservice-name e${e} \
+                       --logfile $PDO_HOME/logs/client.log --loglevel info \
+                       --identity user1 --save-file ${SAVE_FILE} \
+                       "'(inc-value)")
+    if [ $value != $v ]; then
+        die "contract has the wrong value ($value instead of $v) for enclave $e"
+    fi
+done
+
+# commenting out the shell tests for now. 
+:<<'END'
```

Comment:
please uncomment the shell tests in run-tests.sh, they are fine now after the fix

Reply:
done